camel/messages/base.py

```diff
@@ -26,370 +26,306 @@
+        message_type: MessageType = MessageType.DEFAULT,
         meta_dict: Optional[Dict[str, str]] = None,
-        video_bytes: Optional[bytes] = None,
-        image_list: Optional[List[Image.Image]] = None,
-        image_detail: Union[
-            OpenAIVisionDetailType, str
-        ] = OpenAIVisionDetailType.AUTO,
-        video_detail: Union[
-            OpenAIVisionDetailType, str
-        ] = OpenAIVisionDetailType.LOW,
+        subject: Optional[str] = None,
+        reply_with: Optional["BaseMessage"] = None,
+        in_reply_to: Optional["BaseMessage"] = None,
+        envelope: Optional[Envelope] = None,
+        language: str = "en",
+        ontology: Optional[str] = None,
+        protocol: Optional[Dict[str, str]] = None,
     ) -> "BaseMessage":
         return cls(
-            role_name,
-            RoleType.ASSISTANT,
-            meta_dict,
-            content,
-            video_bytes,
-            image_list,
-            OpenAIVisionDetailType(image_detail).value,
-            OpenAIVisionDetailType(video_detail).value,
-        )
-
-    def create_new_instance(self, content: str) -> "BaseMessage":
-        r"""Create a new instance of the :obj:`BaseMessage` with updated
-        content.
-
-        Args:
-            content (str): The new content value.
-
-        Returns:
-            BaseMessage: The new instance of :obj:`BaseMessage`.
-        """
-        return self.__class__(
-            role_name=self.role_name,
-            role_type=self.role_type,
-            meta_dict=self.meta_dict,
+            role_name=role_name,
+            role_type=RoleType.USER,
             content=content,
+            meta_dict=meta_dict,
+            message_type=message_type,
+            sender=sender,
+            receiver=receiver,
+            subject=subject,
+            reply_with=reply_with,
+            in_reply_to=in_reply_to,
+            envelope=envelope,
+            language=language,
+            ontology=ontology,
+            protocol=protocol,
         )
 
-    def __add__(self, other: Any) -> Union["BaseMessage", Any]:
-        r"""Addition operator override for :obj:`BaseMessage`.
-
-        Args:
-            other (Any): The value to be added with.
+    def create_new_instance(self, content: Content) -> "BaseMessage":
+        return self.model_copy(update={"content": content})
 
-        Returns:
-            Union[BaseMessage, Any]: The result of the addition.
-        """
+    def __add__(self, other: Any) -> Union["BaseMessage", Any]:
         if isinstance(other, BaseMessage):
-            combined_content = self.content.__add__(other.content)
-        elif isinstance(other, str):
-            combined_content = self.content.__add__(other)
-        else:
-            raise TypeError(
-                f"Unsupported operand type(s) for +: '{type(self)}' and "
-                f"'{type(other)}'"
+            new_content = Content(
```

Comment:
I think that we could just create a new function in Content class to handle this.